src/modi/mod.rs

```diff
@@ -169,6 +169,20 @@ pub struct LineInfo {
     pub kind: LineInfoKind,
 }
 
+impl LineInfo {
+    pub(crate) fn set_end(&mut self, end_offset: PdbInternalSectionOffset) {
+        debug_assert!(self.offset.section == end_offset.section);
+        if self.offset.section != end_offset.section {
+            return;
+        }
+
+        let length = end_offset.offset - self.offset.offset;
```

Comment:
I get a "subtract with overflow" when iterating on line_program.lines() for this pdb:
https://github.com/mozilla/dump_syms/blob/master/test_data/dump_syms_regtest64.pdb
and this module:
Module { info: DBIModuleInfo { opened: 0, section: DBISectionContribution { section: 1, _padding1: 0, offset: 38556, size: 32, characteristics: 1613770784, module: 27, _padding2: 0, data_crc: 3266334926, reloc_crc: 0 }, flags: 0, stream: StreamIndex(39), symbols_size: 452, lines_size: 0, c13_lines_size: 2304, files: 91, _padding: 0, filename_offsets: 34310784, source: 0, compiler: 0 }, module_name: RawString("f:\\binaries\\Intermediate\\vctools\\crt_bld\\SELF_64_amd64\\crt\\prebuild\\build\\amd64\\mt_obj\\nativec\\\\initcon.obj"), object_file_name: RawString("C:\\Program Files (x86)\\Microsoft Visual Studio 12.0\\VC\\LIB\\amd64\\LIBCMT.lib") }

Reply:
I wonder if this happens due to wrong sorting due to `sections_sorted` as I pointed out above. If the sort order is correct, the only other place where the `end_offset` could potentially be wrong would be:

```
last_line.set_end(section_header.offset + section_header.code_size);
```

for the very last entry.

Reply:
You're correct, it was the sort order because I was sorting by offset rather than section + offset. Reproduced with the above PDB and fixed through always sorting the sections. I'll still guard against this.